src/watchmaker/managers/base.py

```diff
@@ -202,14 +208,27 @@ def call_process(self, cmd):
             stdout=subprocess.PIPE,
             stderr=subprocess.PIPE
         )
-        stdout, stderr = process.communicate()
 
-        for line in stdout.splitlines():
-            self.log.debug('Command stdout: %s', line)
-        for line in stderr.splitlines():
-            self.log.error('Command stderr: %s', line)
+        stdout_reader = threading.Thread(
+            target=self._pipe_logger,
+            args=(process.stdout, self.log.debug, 'Command stdout: '))
+        stdout_reader.daemon = True
+        stdout_reader.start()
+
+        stderr_reader = threading.Thread(
+            target=self._pipe_logger,
+            args=(process.stderr, self.log.error, 'Command stderr: '))
+        stderr_reader.daemon = True
+        stderr_reader.start()
+
+        stdout_reader.join()
+        stderr_reader.join()
+        process.stdout.close()
```

Comment:
I could move the closing of the pipe into `_pipe_logger()`, after the loop, which would better match how the pipe/thread is managed in `subprocess._communicate()`. That way the handle is closed already when the thread returns.